far/vmenu.cpp

```diff
@@ -2011,79 +2097,121 @@ int VMenu::VisualPosToReal(int VPos) const
 	return ItemIterator != Items.cend()? ItemIterator - Items.cbegin() : -1;
 }
 
-size_t VMenu::GetItemMaxShowPos(int Item, const size_t MaxLineWidth) const
+template<typename GetNewHPos_T>
+bool VMenu::SetItemHPos(MenuItemEx& Item, const GetNewHPos_T GetNewHPos, int const TextAreaWidth)
 {
-	const auto Len{ VMFlags.Check(VMENU_SHOWAMPERSAND)? visual_string_length(Items[Item].Name) : HiStrlen(Items[Item].Name) };
-	if (Len <= MaxLineWidth)
-		return 0;
-	return Len - MaxLineWidth;
-}
+	if (Item.Flags & LIF_SEPARATOR) return false;
 
-bool VMenu::SetItemShowPos(int Item, int NewShowPos, const size_t MaxLineWidth)
-{
-	const auto MaxShowPos{ GetItemMaxShowPos(Item, MaxLineWidth) };
-	auto OldShowPos{ Items[Item].ShowPos };
+	const auto ItemLength{ get_item_visual_length(CheckFlags(VMENU_SHOWAMPERSAND), Item.Name) };
+	if (ItemLength <= 0) return false;
 
-	if (NewShowPos >= 0)
+	const auto NewHPos = [&]() -> int
 	{
-		Items[Item].ShowPos = std::min(static_cast<size_t>(NewShowPos), MaxShowPos);
-	}
-	else
-	{
-		Items[Item].ShowPos = MaxShowPos - std::min(static_cast<size_t>(-(NewShowPos + 1)), MaxShowPos);
-	}
+		if constexpr (std::is_invocable_r_v<int, GetNewHPos_T, decltype(Item)>)
```

Comment:
I think you can spell these things more naturally these days, e.g. `if constexpr (requires { GetNewHPos(Item); })` and so on.

Reply:
This would also eliminate the need to define GetNewHPos_T and allow to keep GetNewHPos as auto, making the signature less confusing overall (see my comment above).

Reply:
Thank you for calling it out. Done.